### PR TITLE
Upgrade/element web v1.11.44

### LINKED ISSUE #734: Upgrade to element-web 1.11.44
Diffs : 
element-web : https://github.com/vector-im/element-web/compare/v1.11.39...v1.11.44
react-sdk : 3.78.0->3.81.0 : https://github.com/matrix-org/matrix-react-sdk/compare/v3.78.0...v3.81.0
js-sdk : 27.2.0->28.2.0 : https://github.com/matrix-org/matrix-js-sdk/compare/v27.2.0...v28.2.0

### PR DESCRIPTION
**Pull request by estellecomment**
Fixes #734 

TODO
- [x] fix conflicts
- [x] patches
  - [x] better-help-settings-2 : reimplement it's too complicated to understand.
  - [x] registration-for-mainlining : not sure what happened, probly start over
- [x] ci
  - [x] linter
  - [x] tests : enzyme error. Is it time to remove enzyme like reactsdk ?
  - [x] cypress : settings.spec.ts is broken by translation problem.

Extra things to think about and do
- [x] translations are frozen, halfway through transitioning to another translation system. Are they all broken ? Partly. https://github.com/tchapgouv/tchap-web-v4/issues/703
- [x] translations now are organized into nested categories which is nice. Should we do this too ? Now or later ? -> later. Too much stuff already
- [x] manual tests
  - [x] better-help-settings-2 : run /rageshake and check dialog -> Small regression. Fix it or leave it ?
  - [x] the checklist for version upgrades : https://github.com/tchapgouv/tchap-web-v4/issues/768
- [x] checkboxes in the issue
- [x] couleur des avatars pétée, c'est moins lisible qu'avant -> filed https://github.com/tchapgouv/tchap-web-v4/issues/772, will solve after this PR
- [x] all the languages came back, delete them. They populate the dropdown for chosing the language, it's too noisy. -> added a patch to filter out languages. That way we leave the language files alone.

### CODE REVIEW

**Comment by estellecomment** [comment]
Petite regression du dialog de /rageshake. (est-ce que ca vaut le coup de la fixer ? Ajouté dans la checklist)

Expected : 

image


Obtained : 

image

**Comment by estellecomment** [comment]
couleur des avatars pétée, c'est moins lisible qu'avant. (ajouté à la checklist dans la PR description)

Expected : 

image


Obtained : 

image


Edit : this is filed already in element-web : https://github.com/vector-im/element-web/issues/26285, on 13 sep. No news.

**Comment by estellecomment** [comment]
- [x]    translations : key format has changed, so many translations are broken in our customisations files.

-> filed https://github.com/tchapgouv/tchap-web-v4/issues/769, will fix after this PR is merged.
Edit : actually now its https://github.com/tchapgouv/tchap-web-v4/issues/703

**Comment by estellecomment** [comment]
Translation situation : translation keys are in the process of changing format : "Secure Backup" -> "common|secure_backup". Some keys are already migrated, some are not.

The new format is not custom-translateable with our current setup (tchap-translations.json doesn't work). So for 1.11.44 part of our custom translations disappear. For the next upgrade they will probably all disappear !

Issue already filed, we need to make a module : https://github.com/tchapgouv/tchap-web-v4/issues/703

File: .github/workflows/build_debian.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
             R2_INCOMING_BUCKET: ${{ vars.R2_INCOMING_BUCKET }}
```
Comment (odelcroi) [inline .github/workflows/build_debian.yaml:13]: should we keep this file?

File: .github/workflows/dockerhub.yaml [hunk 2]
```diff
@@ -13,15 +13,15 @@ jobs:
         runs-on: ubuntu-latest
```
Comment (odelcroi) [inline .github/workflows/dockerhub.yaml:13]: should we keep this file?

File: patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch [hunk 3]
```diff
@@ -59,15 +59,14 @@ index 2a0cf58..0ea7031 100644
...
+@@ -730,10 +755,17 @@ export default class CreateSecretStorageDialog extends React.PureComponent<IProp
          if (this.state.phase === Phase.ShowKey) {
              continueButton = (
                  <DialogButtons
--                    primaryButton={_t("Continue")}
--                    disabled={!this.state.downloaded && !this.state.copied && !this.state.setPassphrase}
 +                    // :tchap: change label
 +                    // primaryButton={_t("Continue")}
-+                    primaryButton={_t("I wrote down my code")}
+                     primaryButton={_t("action|continue")}
```
Comment (odelcroi) [inline patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch:67]: not the same label?

File: patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch [hunk 4]
```diff
@@ -185,17 +181,16 @@ index 1de7cc7..7309d2d 100644
...
 --- a/node_modules/matrix-react-sdk/src/toasts/SetupEncryptionToast.ts
 +++ b/node_modules/matrix-react-sdk/src/toasts/SetupEncryptionToast.ts
 @@ -50,7 +50,9 @@ const getIcon = (kind: Kind): string => {
  const getSetupCaption = (kind: Kind): string => {
      switch (kind) {
          case Kind.SET_UP_ENCRYPTION:
--            return _t("Continue");
 +            // :tchap: return _t("Continue");
-+            return _t("Activate");
+             return _t("action|continue");
```
Comment (odelcroi) [inline patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch:196]: should be Activate?

File: .github/workflows/build_debian.yaml (hunk 1 above)
Comment (estellecomment) [reply .github/workflows/build_debian.yaml:13]: No...

I want to move tchap jobs into a separate directory and then just leave all the element files. That way we spend zero time merging files that are deleted or not deleted. https://github.com/tchapgouv/tchap-web-v4/issues/761
Not done yet, I want to keep it separate from a merge PR.

So I'm going to leave them for now.

File: patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch (hunk 3 above)
Comment (estellecomment) [reply patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch:67]: good catch !! 👏  
fixed

File: patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch (hunk 4 above)
Comment (estellecomment) [reply patches/cross-signing-ui/matrix-react-sdk+3.81.0.patch:196]: another good catch !

Ma conclusion c'est qu'on devrait e2e tester les custom translations 🤔  
On pourrait utiliser des snapshots. J'aime pas trop les snapshots parce qu'ils cassent dès que le moindre bout de html change, ce qui arrive hors-bug.
Sinon on code des tests de translations... Ou une sorte de snapshots flexibles, je sais pas si ca existe... 🤔 

Filed https://github.com/tchapgouv/tchap-web-v4/issues/785

**Pull request merged**